src/index.js

```diff
@@ -225,11 +228,17 @@ export default class ReactSearchBox extends Component {
      * The 'value' state is updated with the clicked record's value.
      */
 
-    const { value } = record
+    let {value} = record;
+
+    if(this.props.clearOnSelect) {
+      // clear the value rather than input it into the box
+      value = "";
```

Comment:
You need to modify the state like this:

```suggestion
      this.setState({
        value: ""      
      })
```

Reply:
Apologize as not the greatest at react, but set it to an empty string and the state will be updated below that. If I set the state there value will get set again in the next state call which will cause a race condition. Is that the correct thinking and if so, is there a smarter way to do this than what I'm doing now?